torchdynamo/guards.py

```diff
@@ -28,6 +29,14 @@
 from .utils import tuple_iterator_getitem
 from .utils import tuple_iterator_len
 
+log = logging.getLogger(__name__)
+
+# map from transformed code back to original user code
+orig_code_map = {}
```

Comment:
I'd suggest:
https://github.com/pytorch/torchdynamo/blob/c8e2f7aacd307b76fb294533eddc6b3093603bda/torchdynamo/utils.py#L42

So handle garbage collection a bit better.  This also lets you avoid the calls to `id()`.